knimin/lib/tests/test_data_access.py

```diff
@@ -1451,6 +1457,79 @@ def _basic_test_steps_for_normalized_shotgun_plate(self, before, after,
+               ('iTru5_01_A', 'iTru7_101_03'), ('iTru5_01_A', 'iTru7_101_04'),
+               ('iTru5_01_A', 'iTru7_101_05'), ('iTru5_01_A', 'iTru7_101_06'),
+               ('iTru5_01_A', 'iTru7_101_07'), ('iTru5_01_A', 'iTru7_101_08'),
+               ('iTru5_01_A', 'iTru7_101_09'), ('iTru5_01_A', 'iTru7_101_10')]
+        self.assertEqual(obs, exp)
+
+        # ask for another 10 and make sure it starts where we need to
+        obs = db.generate_i5_i7_indexes('iTru', 10)
+        exp = [('iTru5_01_A', 'iTru7_101_11'), ('iTru5_01_A', 'iTru7_101_12'),
+               ('iTru5_01_A', 'iTru7_102_01'), ('iTru5_01_A', 'iTru7_102_02'),
+               ('iTru5_01_A', 'iTru7_102_03'), ('iTru5_01_A', 'iTru7_102_04'),
+               ('iTru5_01_A', 'iTru7_102_05'), ('iTru5_01_A', 'iTru7_102_06'),
+               ('iTru5_01_A', 'iTru7_102_07'), ('iTru5_01_A', 'iTru7_102_08')]
+        self.assertEqual(obs, exp)
+
+        # just to make sure other 5
+        obs = db.generate_i5_i7_indexes('iTru', 5)
+        exp = [('iTru5_01_A', 'iTru7_102_09'), ('iTru5_01_A', 'iTru7_102_10'),
+               ('iTru5_01_A', 'iTru7_102_11'), ('iTru5_01_A', 'iTru7_102_12'),
+               ('iTru5_01_A', 'iTru7_103_01')]
+        self.assertEqual(obs, exp)
+
+        # we currently have 32256 in the DB and we have requested 25 so
+        # request 32230 and ignore so we can test that the restart happens
+        # correctly
+        db.generate_i5_i7_indexes('iTru', 32230)
+        obs = db.generate_i5_i7_indexes('iTru', 5)
+        exp = [('iTru5_24_H', 'iTru7_114_12'), ('iTru5_01_A', 'iTru7_101_01'),
+               ('iTru5_01_A', 'iTru7_101_02'), ('iTru5_01_A', 'iTru7_101_03'),
+               ('iTru5_01_A', 'iTru7_101_04')]
+        self.assertEqual(obs, exp)
+
+        # test that if we hit the max we start in zero
+        db.generate_i5_i7_indexes('iTru', 32251)
+        obs = db.generate_i5_i7_indexes('iTru', 1)
+        # should be the last of the indices
+        self.assertEqual(obs, [('iTru5_24_H', 'iTru7_114_12')])
+        # should be the first of the indices
+        obs = db.generate_i5_i7_indexes('iTru', 1)
+        self.assertEqual(obs, [('iTru5_01_A', 'iTru7_101_01')])
+
+        # testing errors
+
+        with self.assertRaises(ValueError) as ctx:
+            db.generate_i5_i7_indexes('Should Error', 10)
+        self.assertEqual(
+            ctx.exception.message, "Not a valid shotgun index technology: "
+            "Should Error")
+
+        with self.assertRaises(ValueError) as ctx:
+            db.generate_i5_i7_indexes('BiooNEXTflex-HT', 10)
+        self.assertEqual(
+            ctx.exception.message, "BiooNEXTflex-HT doesn't have any i5 "
+            "values")
+
+        with self.assertRaises(ValueError) as ctx:
+            db.generate_i5_i7_indexes('Nextera', 10)
+        self.assertEqual(
+            ctx.exception.message, "Nextera doesn't have any i7 values")
```

Comment:
Nextera should have both i5 and i7 -- is this an error just because of the test database?